shell/server/core.js

```diff
@@ -147,6 +148,46 @@ class NotificationHandle {
   }
 }
 
+class PersistentSealedUiViewImpl {
+  constructor(token, membraneRequirements) {
+    this._token = token;
+    // TODO: implement SystemPersistent.addRequirements() in a way that collects requirements, and
+    // then saves them appropriately in save().
+    this._membraneRequirements = membraneRequirements || [];
+  }
+
+  save(params) {
+    // Create a new child apiToken for the user whose session attempted to save this capability.
+    // The token should specify this.token as the parent token, and needs to gain an additional membrane
+    // requirement (TODO):
+    // * if params.sealFor is a user, then the user must still have access to the grain that offered
+    //   this cap.  Today, this is handled by offer() in SessionContextImpl.
+    const res = inMeteor(() => {
+      const owner = _.extend({}, params.sealFor);
+      const grainId = this._token.grainId;
+
+      if (owner.user) {
+        // Copy the owner's save label for this capability, if available, to be the new title.
+        if (this._token && this._token.owner && this._token.owner.grain &&
+            this._token.owner.grain.saveLabel && this._token.owner.grain.saveLabel.defaultText) {
+          owner.user.title = this._token.owner.grain.saveLabel.defaultText;
+        }
+
+        // Denormalize this grain's app metadata into owner.user.
+        owner.user.denormalizedGrainMetadata = globalDb.getDenormalizedGrainInfo(grainId);
+      }
+
+      const ret = makeChildTokenInternal(this._token._id, owner, this._membraneRequirements, grainId);
```

Comment:
So, after some discussion with @kentonv and @jparyani, reversing what I said above: 

A SealedUiView *is* actually a UiView.  UiViews are defined to have an additional (implicit) permission, `isHuman`, which human users have, and which is required to make any method calls to the default UiView implementation.  SealedUiView is a child UiView token that has that permission removed.

This means that only `restore()` needs to compute the implicit `mayBeHuman` permission, which is currently always false, since humans do not use the `restore()` function.  This also means that we do not currently require a `sealed` Boolean at all, since we've instead defined it in terms of other properties of the ApiToken - in particular, it's `!!owner.grain`.

This will necessitate some code changes, so I'll do those and get back to you.